---a/net/minecraft/world/gen/structure/MapGenStronghold.java
+++b/net/minecraft/world/gen/structure/MapGenStronghold.java
@@ -1,28 +1,29 @@
 package net.minecraft.world.gen.structure;
 
-import com.google.common.collect.Lists;
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
-import java.util.Random;
 import java.util.Map.Entry;
+import java.util.Random;
+
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.ChunkCoordIntPair;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.BiomeGenBase;
 
-public class MapGenStronghold extends MapGenStructure
-{
-    private List field_151546_e;
+import com.google.common.collect.Lists;
+
+public class MapGenStronghold extends MapGenStructure {
+
+    public final List field_151546_e;
     private boolean ranBiomeCheck;
     private ChunkCoordIntPair[] structureCoords;
     private double field_82671_h;
     private int field_82672_i;
     private static final String __OBFID = "CL_00000481";
 
-    public MapGenStronghold()
-    {
+    public MapGenStronghold() {
         this.structureCoords = new ChunkCoordIntPair[3];
         this.field_82671_h = 32.0D;
         this.field_82672_i = 3;
@@ -31,73 +31,58 @@
         BiomeGenBase[] abiomegenbase = BiomeGenBase.getBiomeGenArray();
         int i = abiomegenbase.length;
 
-        for (int j = 0; j < i; ++j)
-        {
+        for (int j = 0; j < i; ++j) {
             BiomeGenBase biomegenbase = abiomegenbase[j];
 
-            if (biomegenbase != null && biomegenbase.minHeight > 0.0F)
-            {
+            if (biomegenbase != null && biomegenbase.minHeight > 0.0F) {
                 this.field_151546_e.add(biomegenbase);
             }
         }
     }
 
-    public MapGenStronghold(Map p_i2068_1_)
-    {
+    public MapGenStronghold(Map p_i2068_1_) {
         this();
         Iterator iterator = p_i2068_1_.entrySet().iterator();
 
-        while (iterator.hasNext())
-        {
-            Entry entry = (Entry)iterator.next();
+        while (iterator.hasNext()) {
+            Entry entry = (Entry) iterator.next();
 
-            if (((String)entry.getKey()).equals("distance"))
-            {
-                this.field_82671_h = MathHelper.parseDoubleWithDefaultAndMax((String)entry.getValue(), this.field_82671_h, 1.0D);
+            if (((String) entry.getKey()).equals("distance")) {
+                this.field_82671_h = MathHelper.parseDoubleWithDefaultAndMax((String) entry.getValue(), this.field_82671_h, 1.0D);
+            } else if (((String) entry.getKey()).equals("count")) {
+                this.structureCoords = new ChunkCoordIntPair[MathHelper.parseIntWithDefaultAndMax((String) entry.getValue(), this.structureCoords.length, 1)];
+            } else if (((String) entry.getKey()).equals("spread")) {
+                this.field_82672_i = MathHelper.parseIntWithDefaultAndMax((String) entry.getValue(), this.field_82672_i, 1);
             }
-            else if (((String)entry.getKey()).equals("count"))
-            {
-                this.structureCoords = new ChunkCoordIntPair[MathHelper.parseIntWithDefaultAndMax((String)entry.getValue(), this.structureCoords.length, 1)];
-            }
-            else if (((String)entry.getKey()).equals("spread"))
-            {
-                this.field_82672_i = MathHelper.parseIntWithDefaultAndMax((String)entry.getValue(), this.field_82672_i, 1);
-            }
         }
     }
 
-    public String getStructureName()
-    {
+    public String getStructureName() {
         return "Stronghold";
     }
 
-    protected boolean canSpawnStructureAtCoords(int p_75047_1_, int p_75047_2_)
-    {
-        if (!this.ranBiomeCheck)
-        {
+    protected boolean canSpawnStructureAtCoords(int p_75047_1_, int p_75047_2_) {
+        if (!this.ranBiomeCheck) {
             Random random = new Random();
             random.setSeed(this.worldObj.getSeed());
             double d0 = random.nextDouble() * Math.PI * 2.0D;
             int l = 1;
 
-            for (int i1 = 0; i1 < this.structureCoords.length; ++i1)
-            {
-                double d1 = (1.25D * (double)l + random.nextDouble()) * this.field_82671_h * (double)l;
-                int j1 = (int)Math.round(Math.cos(d0) * d1);
-                int k1 = (int)Math.round(Math.sin(d0) * d1);
+            for (int i1 = 0; i1 < this.structureCoords.length; ++i1) {
+                double d1 = (1.25D * (double) l + random.nextDouble()) * this.field_82671_h * (double) l;
+                int j1 = (int) Math.round(Math.cos(d0) * d1);
+                int k1 = (int) Math.round(Math.sin(d0) * d1);
                 BlockPos blockpos = this.worldObj.getWorldChunkManager().findBiomePosition((j1 << 4) + 8, (k1 << 4) + 8, 112, this.field_151546_e, random);
 
-                if (blockpos != null)
-                {
+                if (blockpos != null) {
                     j1 = blockpos.getX() >> 4;
                     k1 = blockpos.getZ() >> 4;
                 }
 
                 this.structureCoords[i1] = new ChunkCoordIntPair(j1, k1);
-                d0 += (Math.PI * 2D) * (double)l / (double)this.field_82672_i;
+                d0 += (Math.PI * 2D) * (double) l / (double) this.field_82672_i;
 
-                if (i1 == this.field_82672_i)
-                {
+                if (i1 == this.field_82672_i) {
                     l += 2 + random.nextInt(5);
                     this.field_82672_i += 1 + random.nextInt(2);
                 }
@@ -109,12 +109,10 @@
         ChunkCoordIntPair[] achunkcoordintpair = this.structureCoords;
         int l1 = achunkcoordintpair.length;
 
-        for (int k = 0; k < l1; ++k)
-        {
+        for (int k = 0; k < l1; ++k) {
             ChunkCoordIntPair chunkcoordintpair = achunkcoordintpair[k];
 
-            if (p_75047_1_ == chunkcoordintpair.chunkXPos && p_75047_2_ == chunkcoordintpair.chunkZPos)
-            {
+            if (p_75047_1_ == chunkcoordintpair.chunkXPos && p_75047_2_ == chunkcoordintpair.chunkZPos) {
                 return true;
             }
         }
@@ -122,18 +122,15 @@
         return false;
     }
 
-    protected List getCoordList()
-    {
+    protected List getCoordList() {
         ArrayList arraylist = Lists.newArrayList();
         ChunkCoordIntPair[] achunkcoordintpair = this.structureCoords;
         int i = achunkcoordintpair.length;
 
-        for (int j = 0; j < i; ++j)
-        {
+        for (int j = 0; j < i; ++j) {
             ChunkCoordIntPair chunkcoordintpair = achunkcoordintpair[j];
 
-            if (chunkcoordintpair != null)
-            {
+            if (chunkcoordintpair != null) {
                 arraylist.add(chunkcoordintpair.getCenterBlock(64));
             }
         }
@@ -141,43 +141,38 @@
         return arraylist;
     }
 
-    protected StructureStart getStructureStart(int p_75049_1_, int p_75049_2_)
-    {
+    protected StructureStart getStructureStart(int p_75049_1_, int p_75049_2_) {
         MapGenStronghold.Start start;
 
-        for (start = new MapGenStronghold.Start(this.worldObj, this.rand, p_75049_1_, p_75049_2_); start.getComponents().isEmpty() || ((StructureStrongholdPieces.Stairs2)start.getComponents().get(0)).strongholdPortalRoom == null; start = new MapGenStronghold.Start(this.worldObj, this.rand, p_75049_1_, p_75049_2_))
-        {
+        for (start = new MapGenStronghold.Start(this.worldObj, this.rand, p_75049_1_, p_75049_2_); start.getComponents().isEmpty() || ((StructureStrongholdPieces.Stairs2) start.getComponents().get(0)).strongholdPortalRoom == null; start = new MapGenStronghold.Start(this.worldObj, this.rand, p_75049_1_, p_75049_2_)) {
             ;
         }
 
         return start;
     }
 
-    public static class Start extends StructureStart
-        {
-            private static final String __OBFID = "CL_00000482";
+    public static class Start extends StructureStart {
 
-            public Start() {}
+        private static final String __OBFID = "CL_00000482";
 
-            public Start(World worldIn, Random p_i2067_2_, int p_i2067_3_, int p_i2067_4_)
-            {
-                super(p_i2067_3_, p_i2067_4_);
-                StructureStrongholdPieces.prepareStructurePieces();
-                StructureStrongholdPieces.Stairs2 stairs2 = new StructureStrongholdPieces.Stairs2(0, p_i2067_2_, (p_i2067_3_ << 4) + 2, (p_i2067_4_ << 4) + 2);
-                this.components.add(stairs2);
-                stairs2.buildComponent(stairs2, this.components, p_i2067_2_);
-                List list = stairs2.field_75026_c;
+        public Start() {
+        }
 
-                while (!list.isEmpty())
-                {
-                    int k = p_i2067_2_.nextInt(list.size());
-                    StructureComponent structurecomponent = (StructureComponent)list.remove(k);
-                    structurecomponent.buildComponent(stairs2, this.components, p_i2067_2_);
-                }
+        public Start(World worldIn, Random p_i2067_2_, int p_i2067_3_, int p_i2067_4_) {
+            super(p_i2067_3_, p_i2067_4_);
+            StructureStrongholdPieces.prepareStructurePieces();
+            StructureStrongholdPieces.Stairs2 stairs2 = new StructureStrongholdPieces.Stairs2(0, p_i2067_2_, (p_i2067_3_ << 4) + 2, (p_i2067_4_ << 4) + 2);
+            this.components.add(stairs2);
+            stairs2.buildComponent(stairs2, this.components, p_i2067_2_);
+            List list = stairs2.field_75026_c;
 
-                this.updateBoundingBox();
-                this.markAvailableHeight(worldIn, p_i2067_2_, 10);
+            while (!list.isEmpty()) {
+                int k = p_i2067_2_.nextInt(list.size());
+                StructureComponent structurecomponent = (StructureComponent) list.remove(k);
+                structurecomponent.buildComponent(stairs2, this.components, p_i2067_2_);
             }
+
+            this.updateBoundingBox();
+            this.markAvailableHeight(worldIn, p_i2067_2_, 10);
         }
-}
+    }
+}
